src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/InMemoryDirectoryInfo.cs

```diff
@@ -17,24 +17,38 @@ public class InMemoryDirectoryInfo : DirectoryInfoBase
     {
         private static readonly char[] DirectorySeparators = new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar };
         private readonly IEnumerable<string> _files;
+        private readonly StringComparison _comparisonType;
 
         /// <summary>
-        /// Creates a new InMemoryDirectoryInfo with the root directory and files given.
+        /// Creates a new case-sensitive InMemoryDirectoryInfo with the root directory and files given.
         /// </summary>
         /// <param name="rootDir">The root directory that this FileSystem will use.</param>
         /// <param name="files">Collection of file names. If relative paths <paramref name="rootDir"/> will be prepended to the paths.</param>
         public InMemoryDirectoryInfo(string rootDir, IEnumerable<string>? files)
-            : this(rootDir, files, false)
+            : this(rootDir, files, false, StringComparison.Ordinal)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new InMemoryDirectoryInfo with the root directory and files given.
+        /// </summary>
+        /// <param name="rootDir">The root directory that this FileSystem will use.</param>
+        /// <param name="files">Collection of file names. If relative paths <paramref name="rootDir"/> will be prepended to the paths.</param>
+        /// <param name="comparisonType">The comparison type for the root directory. When files are enumerated they will be compared with the root directory using this comparison type.</param>
+        public InMemoryDirectoryInfo(string rootDir, IEnumerable<string>? files, StringComparison comparisonType)
```

Comment:
This seems like a reasonable addition to public surface area but it needs to go through [API Review](https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md) for it to be `public`.  Since this is in the same assembly as Matcher, you should use `internal` for now to address your issue.